java/src/jmri/jmrix/tmcc/SerialThrottle.java

```diff
@@ -178,27 +179,39 @@ public void setSpeedSetting(float speed) {
 
             // send to layout option 100 speed steps
         } else if (speedStepMode == jmri.SpeedStepMode.TMCC_100) {
-
-            // TMCC1 ERR 100 speed step mode
+            
+          /** 
+            * TMCC1 ERR 100 speed step mode
+            * purpose is to increase resolution of 32 bits
+            * across 100 throttle 'clicks' by dividing value by 3            
+            * and setting top speed at 32
+          */
             int value = (int) (99 * speed); // max value to send is 99 in 100 step mode
             if (value > 99) {
-                value = 99;    // max possible speed
+                // max possible speed step
+                value = 99;
             }
             SerialMessage m = new SerialMessage();
             m.setOpCode(0xFE);
     
             if (value < 0) {
                 // immediate stop
                 m.putAsWord(0x0060 + address.getNumber() * 128 + 0);
-            } else {
-                // normal speed setting
-                m.putAsWord(0x0060 + address.getNumber() * 128 + value);
             }
+            if (value > 0) {
+                // normal speed step setting
+                m.putAsWord(0x0060 + address.getNumber() * 128 + value / 3);
+            }
+            if (value / 3 > 32) {
+                // max possible speed
+                value / 3 = 32;
+            }
```

Comment:
This section should be above line 201. It currently doesn't do anything.

Also, the line:
`value / 3 = 32;`
should be:
`value = 32 * 3;`